services/api-server/src/simcore_service_api_server/core/application.py

```diff
@@ -96,31 +84,10 @@ def init_app(settings: ApplicationSettings | None = None) -> FastAPI:
     app.add_event_handler("startup", create_start_app_handler(app))
     app.add_event_handler("shutdown", create_stop_app_handler(app))
 
-    app.add_exception_handler(HTTPException, http_error_handler)
-    app.add_exception_handler(HttpxException, handle_httpx_client_exceptions)
-    app.add_exception_handler(RequestValidationError, http422_error_handler)
-    app.add_exception_handler(LogDistributionBaseException, log_handling_error_handler)
-    app.add_exception_handler(CustomBaseError, custom_error_handler)
-
-    # SEE https://docs.python.org/3/library/exceptions.html#exception-hierarchy
-    app.add_exception_handler(
-        NotImplementedError,
-        make_http_error_handler_for_exception(
-            NotImplementedError,
-            status.HTTP_501_NOT_IMPLEMENTED,
-            detail_message="Endpoint not implemented",
-        ),
-    )
-    app.add_exception_handler(
-        Exception,
-        make_http_error_handler_for_exception(
-            Exception,
-            status.HTTP_500_INTERNAL_SERVER_ERROR,
-            detail_message="Unexpected error",
-            add_exception_to_message=(settings.SC_BOOT_MODE == BootModeEnum.DEBUG),
-            add_oec_to_message=True,
-        ),
+    exceptions.setup_exception_handlers(
+        app, is_debug=settings.SC_BOOT_MODE == BootModeEnum.DEBUG
```

Comment:
Personally I find it more readable if there are parentheses in this case. Just a suggestion...

Reply:
I wanted to avoid making setup depending on settings.